### PR TITLE
Fix remote field

### PR DESCRIPTION
**Pull request by almasaeed2010**
Issue #551

## Type(s) of Change(s)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API-specific change (fix or addition to an API function)
- [ ] Updates documentation (inline or markdown files)

## Description

The remote data field needed some error checking to avoid usage of undefined variables.

## Testing?

Unfortunately I was not able to replicate the issue from scratch so I am not sure how to test this. I do think though that a quick look at the code should suffice since all we are adding is simple error checking and provide a fallback when a variable or a value is missing.

## Additional Notes (if any):

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by bradfordcondon** [comment]
these changes look really no-nonsense to me, just checking if things are set. Approved.

note: the whitespace changes make this look like more than it is: they can be disabled n the PR review

File: tripal_ws/includes/TripalFields/remote__data/remote__data.inc [hunk 1]
```diff
@@ -237,6 +237,11 @@ class remote__data extends WebServicesField {
     if (preg_match('/\?/', $query)) {
       list($path, $q) = explode('?', $query);
     }
+
+    if(empty($remote_site)) {
+      return null;
```
Comment (laceysanderson) [inline tripal_ws/includes/TripalFields/remote__data/remote__data.inc]: I would add a tripal_report_error() before returning NULL. That way an admin would see there's a problem if looking in the recent messages log rather then left scratching their head. :-)

File: tripal_ws/includes/TripalFields/remote__data/remote__data.inc [hunk 2]
```diff
@@ -245,14 +250,24 @@ class remote__data extends WebServicesField {
...
    private function makeRemoteRequest($query) {
      $path = $query;
      $q = '';
      if (preg_match('/\?/', $query)) {
        list($path, $q) = explode('?', $query);
      }
-     $data = tripal_get_remote_content($this->remote_site->id, $path, $q);
+
+    if(empty($this->remote_site)) {
+      return null;
```
Comment (laceysanderson) [inline tripal_ws/includes/TripalFields/remote__data/remote__data.inc]: Always use tripal_report_error() when returning null.

File: tripal_ws/includes/TripalFields/remote__data/remote__data.inc [hunk 3]
```diff
@@ -245,14 +250,24 @@ class remote__data extends WebServicesField {
...
+    }
+
+     try {
+       $data = tripal_get_remote_content($this->remote_site->id, $path, $q);
+     } catch (Exception $exception) {
+       return null;
```
Comment (laceysanderson) [inline tripal_ws/includes/TripalFields/remote__data/remote__data.inc]: Always use tripal_report_error() when returning null.

File: tripal_ws/includes/TripalFields/remote__data/remote__data_formatter.inc [hunk 4]
```diff
@@ -139,19 +139,23 @@ class remote__data_formatter extends WebServicesFieldFormatter {
...
     $content .= '<p>';
 
-    $content .=  t('This content provided by !site.',
-      array('!site' => l($site->name, $site->url, array('attributes' => array("target" => '_blank')))));
+    if($site) {
+      $content .= t('This content provided by !site.', [
+          '!site' => l($site->name, $site->url,
+            ['attributes' => ["target" => '_blank']])
+        ]);
+    }
```
Comment (laceysanderson) [inline tripal_ws/includes/TripalFields/remote__data/remote__data_formatter.inc:150]: I would also add an else with a tripal_report_error() here.

**Review by laceysanderson — COMMENTED** [review]
I agree the code looks pretty straightforward :-) My only comment is that it could benifit from usage of `tripal_report_error()` so that admin running into these edge cases would see what the problem is when looking at the logs. :-)

Thanks for the PR! :-)

**Comment by almasaeed2010** [comment]
Hello @laceysanderson 

Thanks for the suggestions!

I added error reporting although I am not sure the messages are very helpful since I am not entirely positive which situations we'd have a missing site and still get to that part of the code. In my opinion, we should figure out a way to null the field in the loader before encountering this part of the code or report the error only once at the top of the loader then exiting since there is no need to continue if the site is not available.

Any thoughts? I could reimplement this PR to issue 1 warning instead 3 in the loader if you agree it's a better way.

Thanks!

**Review by laceysanderson — APPROVED** [review]
I’m content with this solution 🙂 Thanks @almasaeed2010 for the PR and the fixes!

**Pull request merged**